include/zephyr/drivers/sensor_types.h

```diff
@@ -0,0 +1,232 @@
+/**
+ * @file drivers/sensor_types.h
+ *
+ * @brief Public APIs for the sensor driver.
+ */
+
+/*
+ * Copyright (c) 2022 Google Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef INCLUDE_DRIVERS_SENSOR_TYPES_H_
+#define INCLUDE_DRIVERS_SENSOR_TYPES_H_
+
+#include <zephyr/math/util.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @file
+ * The SENSOR_TYPE_* defines are the sensor types supported. Additional types can be added at
+ * SENSOR_TYPE_VENDOR_START.
+ */
+
+/**
+ * Start value for all of the vendor-defined private sensors.
+ */
+#define SENSOR_TYPE_VENDOR_START UINT8_C(192)
+
+/**
+ * Accelerometer sensor returning data in m/s^2.
+ */
+#define SENSOR_TYPE_ACCELEROMETER UINT8_C(1)
+
+/**
+ * Instantaneous motion detection.
+ *
+ * This is a one-shot sensor.
+ */
+#define SENSOR_TYPE_INSTANT_MOTION_DETECT UINT8_C(2)
+
+/**
+ * Stationary detection.
+ *
+ * This is a one-shot sensor.
+ */
+#define SENSOR_TYPE_STATIONARY_DETECT UINT8_C(3)
+
+/**
+ * Gyroscope sensor returning data in radians / sec
+ */
+#define SENSOR_TYPE_GYROSCOPE UINT8_C(6)
+
+/**
+ * Magnetometer sensor returning data in Teslas.
+ */
+#define SENSOR_TYPE_GEOMAGNETIC_FIELD UINT8_C(8)
+
+/**
+ * Barometric pressure sensor returning data in Pascals.
+ */
+#define SENSOR_TYPE_PRESSURE UINT8_C(10)
+
+/**
+ * Ambient light sensor returning data in lux.
+ *
+ * This is an on-change sensor.
+ */
+#define SENSOR_TYPE_LIGHT UINT8_C(12)
+
+/**
+ * Proximity detection.
+ *
+ * This is an on-change sensor.
+ */
+#define SENSOR_TYPE_PROXIMITY UINT8_C(13)
+
+/**
+ * Step detection.
+ */
+#define SENSOR_TYPE_STEP_DETECT UINT8_C(23)
+
+/**
+ * Step counter.
+ *
+ * This is an on-change sensor.
+ */
+#define SENSOR_TYPE_STEP_COUNTER UINT8_C(24)
+
+/**
+ * Accelerometer temperature sensor returning the accelerometer's temperature in Kelvin.
+ */
+#define SENSOR_TYPE_ACCELEROMETER_TEMPERATURE UINT8_C(56)
+
+/**
+ * Gyroscope temperature sensor returning the gyroscope's temperature in Kelvin.
+ */
+#define SENSOR_TYPE_GYROSCOPE_TEMPERATURE UINT8_C(57)
+
+/**
+ * Magnetometer temperature sensor returning the magnetometer's temperature in Kelvin.
+ */
+#define SENSOR_TYPE_GEOMAGNETIC_FIELD_TEMPERATURE UINT8_C(58)
+
+BUILD_ASSERT(SENSOR_TYPE_GEOMAGNETIC_FIELD_TEMPERATURE < SENSOR_TYPE_VENDOR_START,
+	     "Too many sensor types");
+
+struct sensor_data_header {
+	/** The number of allocated `readings` elements. */
+	const uint16_t reading_size;
+
+	/** The base timestamp, in nanoseconds. */
+	uint64_t base_timestamp;
+
+	/** The number of valid elements in the 'readings' array. */
+	uint16_t reading_count;
```

Comment:
move this attribute at 2nd place to make sure compiler is not going to add padding for alignment which will bloat the structure.